zk.js/src/wallet/user.ts

```diff
@@ -1610,10 +1591,10 @@ export class User {
   }) {
     const programAddress = getVerifierProgramId(idl);
     const balance = latestBalance
-      ? await this.syncStorage(idl, true, merkleTree)
+      ? await this.syncStorage(idl, true)
```

Comment:
Why did you remove this?
It doesn't matter because we are going to delete this code anyway, but in general on some rpc request level you will want the option to provide a Merkle tree set public key so that you define the search space you want to get your utxo from.

Reply:
Since `Rpc`/`TestRpc` is being passed to the `User` class, and it already contains the information about which MT is being used, passing it again in random parts of the class just seemed redundant for me. This way made fixing the merkle_tree_tests easier - I'm generating a new MTS keypair there and passing it once to TestRpc was just easier than passing it over in multiple places.

Before my changes, this `merkleTree` variable was set to `MerkleTreeConfig.getTransactionMerkleTreePda()` in some other random methods of `User` - `syncStorate`, `getUtxos` etc. I don't find this intuitive. I rather prefer to set the MT pubkey for the whole class once.

The other option would be to have it in the constructor.

Anyways, this is going to be removed, so maybe that doesn't matter.